assets/src/data/eventespresso/schema/resolvers.js

```diff
@@ -134,6 +136,37 @@ export function* getRelationEndpointForEntityId(
 	return endpoint;
 }
 
+/**
+ * A resolver for getting the primary key string to use in a query for the given
+ * model and relation. This considers the join type for the relation.
+ *
+ * @see the `getRelationPrimaryKeyString` selector for example.
+ *
+ * @param {string} modelName
+ * @param {string} relationName
+ * @return {string} The primary key string to use or an empty string if relation
+ * type could not be determined.
+ */
+export function* getRelationPrimaryKeyString( modelName, relationName ) {
+	// normalize
+	modelName = singularModelName( modelName );
+	relationName = pluralModelName( relationName );
+	const relationType = yield resolveSelect(
+		SCHEMA_REDUCER_KEY,
+		'getRelationType',
+		modelName,
+		relationName
+	);
+	relationName = singularModelName( relationName );
+	if ( relationType === '' ) {
+		return '';
+	}
```

Comment:
since `relationName` is not needed yet, maybe bump its creation below this return clause.
ie:

```js
	if ( relationType === '' ) {
		return '';
	}
	relationName = singularModelName( relationName );
```